go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -202,6 +202,9 @@ func ConvertPodFailureToError(status v1.PodStatus) (code, message string) {
 					c.LastTerminationState.Terminated.ExitCode,
 					c.LastTerminationState.Terminated.Reason,
 					c.LastTerminationState.Terminated.Message)
+			} else if c.State.Terminated != nil && strings.Contains(c.State.Terminated.Reason, OOMKilled) {
```

Comment:
Seems like we use this logic a lot (also in the method `DemystifySuccess`), perhaps we can encapsulate it in a method like `isOOMKilled()`?

Reply:
hm i'm not a huge fan of refactoring one liners (especially using haytham's suggestion below) but let me know if you feel strongly